configs/baremetal.h

```diff
@@ -116,6 +116,8 @@
     MBEDTLS_SSL_EXTENDED_MS_ENFORCE_ENABLED
 
 #define MBEDTLS_USE_TINYCRYPT
+#define MBEDTLS_HAVE_ASM
```

Comment:
@AndrzejKurek , why do we need MBEDTLS_HAVE_ASM flag in baremetal?

Reply:
I think it is already defined in main config and we should not touch it ...

Reply:
What do you mean by the "main config"? I assumed that we want to have the ASM optimization on at all times, so ASM has to be enabled. I can turn the logic around if you want, to check if ASM is enabled, and if so, then turn on the ASM optimization instead.

Reply:
I mean that in this file https://github.com/ARMmbed/mbedtls/blob/baremetal/include/mbedtls/config.h#L61 it is enabled by default. 
Actually I am asking why do we want to perform coupling between  2 flags? Why only we can not switch on only MBEDTLS_OPTIMIZE_TINYCRYPT_ASM ?

Reply:
Because it won't work without tinycrypt and MBEDTLS_HAVE_ASM turned on.